fraud/sync.go

```diff
@@ -0,0 +1,143 @@
+package fraud
+
+import (
+	"context"
+	"time"
+
+	"github.com/ipfs/go-datastore"
+	"github.com/libp2p/go-libp2p-core/event"
+	"github.com/libp2p/go-libp2p-core/network"
+	"github.com/libp2p/go-libp2p-core/peer"
+	pubsub "github.com/libp2p/go-libp2p-pubsub"
+
+	"github.com/celestiaorg/go-libp2p-messenger/serde"
+
+	pb "github.com/celestiaorg/celestia-node/fraud/pb"
+)
+
+// syncFraudProofs encompasses the behavior for fetching fraud proofs from other peers.
+// syncFraudProofs subscribes to EvtPeerIdentificationCompleted in order to get newly connected peers
+// and request fraud proofs from them.
+// Once fraud proofs are received, they will be published across all local subscriptions in
+// order to be verified.
+func (f *ProofService) syncFraudProofs(ctx context.Context) {
+	log.Debug("start fetching Fraud Proofs")
+	sub, err := f.host.EventBus().Subscribe(&event.EvtPeerIdentificationCompleted{})
+	if err != nil {
+		log.Error(err)
+		return
+	}
+	defer sub.Close()
+	f.topicsLk.RLock()
+	proofTypes := make([]uint32, 0, len(f.topics))
+	for proofType := range f.topics {
+		proofTypes = append(proofTypes, uint32(proofType))
+	}
+	f.topicsLk.RUnlock()
+	connStatus := event.EvtPeerIdentificationCompleted{}
+	for i := 0; i < fraudRequests; i++ {
+		select {
+		case <-ctx.Done():
+			return
+		case e := <-sub.Out():
+			connStatus = e.(event.EvtPeerIdentificationCompleted)
+		}
+
+		if connStatus.Peer == f.host.ID() {
```

Comment:
Is this what this if does?
```suggestion
		// ignore ourselves as a peer
		if connStatus.Peer == f.host.ID() {
```

Reply:
:yes: